internal/runners/pull/pull.go

```diff
@@ -272,7 +272,18 @@ func (p *Pull) mergeBuildScript(remoteCommit, localCommit strfmt.UUID) error {
 	}
 
 	// Write the merged build expression as a local build script.
-	return buildscript.Update(p.project, atTime, mergedExpr, p.auth)
+	err = buildscript.Update(p.project, atTime, mergedExpr, p.auth)
+	if err != nil {
+		return "", errs.Wrap(err, "Could not update build script with merged changes")
+	}
+
+	if synced, err := buildscriptRunbits.Sync(p.project, nil, p.out, p.auth); err == nil && synced {
+		return localcommit.Get(p.project.Dir())
+	} else {
+		return "", errs.Wrap(err, "Could not commit merged build script")
+	}
```

Comment:
This change here may cause this ticket to be blocked on https://activestatef.atlassian.net/browse/DX-2619, where we want to do away with `buildscript.Sync()`.